### DISCUSSION TITLE
space-y upgrade

### DISCUSSION THREAD

**Discussion by federicocappellotto97** [tailwindlabs/tailwindcss#14322] [Ideas]
I would like to suggest an upgrade to the classes space-y and space-x, useful to override the margin on a single element
I use to define it like this:
```
.space-y-custom > * ~ * {
    @apply mt-[var(--mt)];
}
```

Then i use it like this:

```
<ul class="space-y-custom [--mt:theme(spacing.40)]"> //all the items will have margin-top: 40px
    <li>Item 1</li>
    <li>Item 2</li>
    <li>Item 3</li>
    <li class="[--mt:theme(spacing.120)]">Item 4</li> //only this item will have margin-top: 120px
    <li>Item 5</li>
</ul>
```

**Comment by MaxLap** [comment]
I wonder if instead, there could be a distinct CSS layer for "mass-utilities" like `space-y`.

Conceptually, I consider an utility like "space-y" to be a custom default value we want for the children. While I consider "mt" to be a specific value that we want in a specific location.

But since the selector for `space` is more specific than the one for margins such as `mt`, `mt` gets ignored.

In my solution, "mass-utilities" would be a different layer (earlier than "utilities"), and therefore would be overwritten by regular utilities.

I feel this behavior would be a lot more intuitive andn flexible without needing tons of variables for every variations.

"mass-utilities" may not be the best name, maybe "delegated-utilities" or "descendant-utilities"?

Since 4.0 is being worked on right now, this would be a great time to add this feature since this is technically a breaking change if someone had a useless `mt-10` within his `space-y-4`.

**Reply by federicocappellotto97** [reply]
Yes, even this is a good idea!

**Comment by MaxLap** [comment]
Interesting, reading this https://tailwindcss.com/blog/tailwindcss-v4-alpha, I learned that Tailwind doesn't use actual CSS layers! 4.0 is changing that, so what I suggested wasn't possible before.

@adamwathan: Any thoughts on this distinct "mass-utilities" CSS layer idea?

**Reply by adamwathan** [reply]
In v4 we’ve reduced the specificity of utilities like space-y to zero so you can override it on child elements no problem now.
